### DISCUSSION TITLE
Clickable tiles

### DISCUSSION THREAD

**Discussion by dystechnic** [Daveiano/weewx-wdc#211] [Ideas]
Hi,

As I mentioned before,we are trying to make an informational touchscreen at our observatory using your skin. We are making serious progress with displaying data from our Davis weather station and our all-sky cams . But because we get are a lot of school classes visiting our observatory we also get a lot of questions about what they see on the display. 
Wouldn't it be nice if the kids could ' click'  on a tile and then get some explanation about what the see? For example the term 'Evapotranspiration'. If the touch (click) on that tile there could be a redirect to a page that explains what that means. This way they get a bit of education while they are playing with the screen.

Is it possible to do this? Is this already possible maybe?

Kind Regards,

Ed.

**Comment by Daveiano** [comment]
Hi Ed,

interesting idea! No, sadly this is not yet possible (without modifying the templates).

>  If the touch (click) on that tile there could be a redirect to a page that explains what that means.

What do you mean by this? Simply the tiles are linked to a page, eg. Wikipedia that opens in a new tab? This would be easy to implement, I guess.

I also thought about Tooltips, these can also be configured to be triggered via click/touch. The "Time" Table header uses one to explain the date format used in the table:

![GH-211](https://github.com/Daveiano/weewx-wdc/assets/3184897/4be9b9d9-735a-4462-858d-d5a760d2bc08)

But this would only work for short descriptions.

**Comment by dystechnic** [comment]
Hi,

> What do you mean by this? Simply the tiles are linked to a page, eg. Wikipedia that opens in a new tab? This would be easy to implement, I guess.

Yes, that's what I mean. But not to Wikipedia but to a local page we create for  the specific topic. If it opens in a seperate tab that would be great.
Not all tiles need explanations so if it is optional than that would be perfect. Otherwise we can let unused tiles just point to the index page. In case someone clicks it nothing breaks.
> 
> I also thought about Tooltips, these can also be configured to be triggered via click/touch. The "Time" Table header uses one to explain the date format used in the table:
> 
Tooltips on tiles is also a great idea. But hover functions do not work on a touchscreen so that doesn't work in this particular case.

Regards,

Ed.

**Comment by Daveiano** [comment]
Ok, that's easy to implement and should find its way into the upcoming release. 

I thought about a little indicator for the tile to show that's its clickable, like this:

![GH-211-1](https://github.com/Daveiano/weewx-wdc/assets/3184897/8086ba5b-66dc-48da-9795-d3af8a6a305c)

In this example the Outside Temp tile is clickable. What do you think about that?

Do you only need this for the Stat tiles or also for the diagram tiles?

**Reply by dystechnic** [reply]
Hi,

Yes, that's a perfect solution.! 
In this example, is it selectable which tiles are clickable? And can they all point to a different page where we can put our own info? I was thinking a bit like the about page.

I did some experimenting on the screen yesterday and it works best if the reffered page is openened in the same window or tab as the tiles . So not in a new tab. Does that sound logical?

Sorry for all the questions.
Regards,
Ed.

**Reply by Daveiano** [reply]
Yes, you can define which tiles and you can define which link per tile, the updated wiki will sound like that:

> `[[stat_tile_links]]` Optionally you can link tiles to eg. Explanation pages, this means you can make the tiles 'clickable'. For example, if somebody wants to link the `ET` tile to the corresponding wiki page, something like this should work:
>
> ```
> [DisplayOptions]
>     ...
>     [[stat_tile_links]]
>         [[[ET]]]
>             link = "https://en.wikipedia.org/wiki/Evapotranspiration"
>             open_new_tab = True
> ```

I also added `open_new_tab` as a configuration, so you can change it easily.

I will add this to the next 3.4.0 release, feel free to comment if you want to see more features ;)

**Reply by dystechnic** [reply]
Thank you! I can't wait to try this.
Just out of curiousity, When will 3.4.0 be released?

**Reply by Daveiano** [reply]
Probably within this week, if not next week ;)

**Reply by Daveiano** [reply]
@dystechnic Release is out: https://github.com/Daveiano/weewx-wdc/releases/tag/v3.4.0 

You can download and test the new features, NL translation and changeable icon is also included.

**Reply by dystechnic** [reply]
Thank you.!! 
I will install it on my test environment tomorrow.

**Comment by dystechnic** [comment]
The clickable tiles work verry well!  Thank you for that.
But I have yet another  small feature request. Would it be possible to open the destination sites or HTML pages in the main I-frame of the skin in stead of in a full browser window?
![weewx-wdc](https://github.com/Daveiano/weewx-wdc/assets/36402924/82e1387f-553a-4a8d-93e6-e95c9f494ce4)

I mean the yellow part of the site in attached screenshot.

Again, thank you for your effort and the quick responses.
Regards,
Ed.

**Reply by Daveiano** [reply]
Please have a look at #220, this discussion should be solved. I am glad the clickable tiles work for you!
